main.cpp

```diff
@@ -0,0 +1,95 @@
+#include <iostream>
+#include <vector>
+#include <string>
+#include <sstream>
+#include <unistd.h>
+#include <sys/wait.h>
+#include <cstdlib>
+#include <fstream>
+using namespace std ;
+
+ofstream history_file("history.txt", ios::app);
```

Comment:
All the special strings should be in consts for better readability and understanding of the code.